Examples/AdvancedImageReading/Documentation.rst

```diff
@@ -0,0 +1,60 @@
+.. _lbl_advanced_image_reading:
+
+Advanced Image Reading
+==============================
+
+Overview
+--------
+
+This example illustrates advanced image reading options:
+  1. Explicitly selecting a specific IO for reading.
+  2. Querying an image for its meta-data without reading the bulk intensity data.
+  3. Reading a sub-region of an image.
+
+Selecting a Specific IO
+++++++++++++++++++++++++
+When using the default settings for the `ImageFileReader class <https://itk.org/SimpleITKDoxygen/html/classitk_1_1simple_1_1ImageFileReader.html>`_ or
+the `ReadImage <https://itk.org/SimpleITKDoxygen/html/namespaceitk_1_1simple.html#ae3b678b5b043c5a8c93aa616d5ee574c>`_ function
+you have minimal control over the reading. That is, the specific IO mechanism is
+determined automatically based on the file type and the whole file is read into
+memory.
+
+In some cases there are multiple IO classes that support reading the same image
+format and you do not know which one was used. For example the 'SCIFIOImageIO'
+and 'JPEGImageIO' both support reading JPEG images. The SCIFIOImageIO reading
+is much slower than the JPEGImageIO. With the advanced reading approach you can
+find out which IO classes are supported by the ImageFileReader and indicate which
+one you want to use.
+
+Querying an image for its meta-data
+++++++++++++++++++++++++++++++++++++
+
+In some cases you may only want to read an image's meta-data without reading the
+bulk intensity data into memory. For instance, if you want to display image
```

Comment:
For this particular case we need to know the image's size before we do sub-region IO.

Please reference the DICOM meta-dictionary example which already does this... stay focused on the region IO.